### PR TITLE
PORTALS-2580 - refactor for synapse-oauth-signin + misc fixes

### PR DESCRIPTION
**Pull request by nickgros**
Changes I've made in synapse-oauth-signin are dependent upon the changes to useDetectSSOCode, which is how this work is related to PORTALS-2580

-----

- Refactor useDetectSSOCode so SSO redirect logic can be captured in-app
- Upgrade mui-one-time-password-input to 1.1.0 and use new `autoFocus`
- Fix dialog UI in portals

-----

Current portals login dialog needs more padding: 

image

after changes:
image

### CODE REVIEW

File: packages/synapse-react-client/src/lib/utils/hooks/useDetectSSOCode.ts [hunk 1]
```diff
@@ -53,26 +52,17 @@ export default function useDetectSSOCode(
...
     if (code && provider) {
       const redirectUrl = `${redirectURL}?provider=${provider}`
-      const redirectAfterSuccess = () => {
-        // go back to original route after successful SSO login
-        const originalUrl = localStorage.getItem('after-sso-login-url')
-        localStorage.removeItem('after-sso-login-url')
-        if (originalUrl) {
-          window.location.replace(originalUrl)
-        }
-      }
```
Comment (nickgros) [inline packages/synapse-react-client/src/lib/utils/hooks/useDetectSSOCode.ts:63]: This redirect logic seems to me like it should be app-specific, so it's now an argument `onSignInComplete`

File: packages/synapse-react-client/src/lib/utils/hooks/useDetectSSOCode.ts [hunk 2]
```diff
@@ -16,12 +16,10 @@ import { TwoFactorAuthErrorResponse } from '../synapseTypes/ErrorResponse'
 import { PROVIDERS } from '../../containers/auth/AuthenticationMethodSelection'
 
 type UseDetectSSOCodeOptions = {
+  onSignInComplete?: () => void
   registerAccountUrl?: string
   onError?: (err: unknown) => void
-  onTwoFactorAuthRequired?: (
-    resp: TwoFactorAuthErrorResponse,
-    callback: () => void,
```
Comment (nickgros) [inline packages/synapse-react-client/src/lib/utils/hooks/useDetectSSOCode.ts:23]: This `callback` parameter is unused, I think it was mistakenly thrown in when I originally refactored this.

File: apps/portals/src/utils.ts [hunk 3]
```diff
@@ -12,3 +12,24 @@ export const DESKTOP_VIEWPORT_MIN_WIDTH_MEDIA_QUERY = `(min-width: ${DESKTOP_VIE
 
 export const useShowDesktop = () =>
   useMedia(DESKTOP_VIEWPORT_MIN_WIDTH_MEDIA_QUERY)
+
```
Comment (nickgros) [inline apps/portals/src/utils.ts:15]: May move these functions back into SRC or a different app utils project when we have multiple apps in the monorepo and we want to refactor duplicate code

File: apps/portals/craco.config.js [hunk 4]
```diff
@@ -2,19 +2,26 @@ const { ProvidePlugin } = require('webpack')
...
-        },
-      },
-      ignoreWarnings: [/Failed to parse source map/],
+    configure: (webpackConfig) => {
+      const scopePluginIndex = webpackConfig.resolve.plugins.findIndex(
+        ({ constructor }) =>
+          constructor && constructor.name === 'ModuleScopePlugin',
+      )
+
+      webpackConfig.resolve.plugins.splice(scopePluginIndex, 1)
```
Comment (nickgros) [inline apps/portals/craco.config.js:11]: Ran into a strange issue where a module within `@babel/runtime` was being imported from the root `node_modules` and it didn't recognize it as an external dependency, so the `ModuleScopePlugin` failed saying that the imported module wasn't in `./src`. 

It's not clear to me what change actually caused this issue to surface.

I couldn't find a real fix for this other than removing the plugin. This seemed to not cause problems for now, and I want to replace webpack with vite soon, so I think this is an acceptable stopgap.

File: apps/portals/src/Navbar.tsx [hunk 5]
```diff
@@ -212,38 +214,38 @@ class Navbar extends React.Component<any, State> {
                   >
                     SIGN&nbsp;IN
                   </Button>
-                  <Modal
-                    animation={false}
-                    backdrop="static"
-                    keyboard={false}
-                    onHide={() => {
+                  <Dialog
```
Comment (nickgros) [inline apps/portals/src/Navbar.tsx:217]: Use @mui instead of react-bootstrap

File: apps/portals/craco.config.js (hunk 4 above)
Comment (jay-hodgson) [inline apps/portals/craco.config.js:11]: Why does the ModuleScopePlugin need to be removed?

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**